engine/src/test/java/org/hibernate/search/test/configuration/sharding/ShardingConfigurationTest.java

```diff
@@ -262,6 +266,21 @@ private MutableSearchFactory getSearchFactory(Map<String, String> shardingProper
 		).buildSearchIntegrator();
 	}
 
+	private static Path getTargetDir() {
+		URI classesDirUri;
+		try {
+			classesDirUri = ShardingConfigurationTest.class.getProtectionDomain()
+					.getCodeSource()
+					.getLocation()
```

Comment:
I'm not familiar with this API. This would - like the existing test helper - attempt to write into the JAR's URL when referring to the class of a module which isn't the one currently being compiled by Maven right?

If so, what's the benefit in changing this from using the current helper?